[lint] Remove Fusesoc-related message waivers

msfschaffner opened the pull request:
This can be removed due to #1033.

Signed-off-by: Michael Schaffner <msf@opentitan.org>

rswarbrick commented:
This seems to be failing CI because of the alert naming scheme?? I don't really understand why this change could trigger that, though, so maybe there's something more interesting going on?

msfschaffner commented:
From looking at the CI logs, it looks like topgen is run under the hood for DV targets, and these new alert warnings leak out into the lint tool report, thereby causing style lint to fail.

Maybe it makes sense to suppress the topgen warnings when calling DV targets?

msfschaffner commented:
Ok I took a closer look and modified the lint class as follows:
- first, all the "tool" warnings and errors are actually "flow" warnings and errors, since they are parsed out of the fusesoc log. hence I renamed them accordingly in the lint parser scripts. 
- the regex I've removed in this PR apparently masked out some other flow warnings from topgen. I reduced the severity of flow warnings such that they do not block PR CI anymore. there are more warnings than just the alert name mismatches, and it will be difficult to address them all at this point.
- also, I've noticed that some properties in FPV code leads to Verible syntax errors. I've provisioned an error message filter for these, but we need to wait for https://github.com/google/verible/issues/652 to be addressed first before we can actually activate it. tracked in issue https://github.com/lowRISC/opentitan/issues/4978.

rswarbrick commented on hw/lint/tools/ascentlint/parse-lint-report.py:
```diff
@@ -111,13 +111,14 @@ def main():
-        sys.exit(1)
+    # Pass/fail status is determined in the LintCfg class.
+    log.info(("Found %d flow warnings, %d flow errors, %d lint infos,\n"
+             "%d lint warnings and %d lint errors."),
```
Nit: Missing space to align the second string literal

msfschaffner replied:
fixed, thanks

rswarbrick commented on hw/lint/tools/ascentlint/parse-lint-report.py:
```diff
@@ -111,13 +111,14 @@ def main():
-    # lint infos do not count as failures
-    n_errors = len(results["errors"]) + len(results["lint_errors"])
-    n_warnings = len(results["warnings"]) + len(results["lint_warnings"])
-    if n_errors > 0 or n_warnings > 0:
-        log.info("Found %d lint errors and %d lint warnings", n_errors, n_warnings)
-        sys.exit(1)
```
Don't we still want to exit with status 1 if `results['errors']` or `results['warnings']` is positive?

msfschaffner replied:
I removed it since the error status of Dvsim is not determined here, and I sort of wanted to avoid that we categorize the message severities in multiple places...
But I can move it back in if you think it would be good to have.

rswarbrick replied:
Oh, I see. Yep, fine with me.

msfschaffner commented:
Thanks @rswarbrick for the quick review. Can you push the magic "approve" button so I can merge this in ;)?

rswarbrick reviewed (approved):
Oops, sorry!
